src/python/pants/backend/python/tasks2/resolve_requirements_task_base.py

```diff
@@ -0,0 +1,148 @@
+from pants.task.task import Task
+from pants.util.dirutil import safe_rmtree
+
+
+class PythonRequirementFingerprintStrategy(DefaultFingerprintHashingMixin,
+                                           TaskIdentityFingerprintStrategy):
+
+  def compute_fingerprint(self, req_lib):
+    hash_elements_for_target = []
+    hash_elements_for_target.extend([req.cache_key() for req in req_lib.requirements])
+    if not hash_elements_for_target:
+      return None
+    hasher = self._build_hasher(req_lib)
+    for element in hash_elements_for_target:
+      hasher.update(element)
+    return hasher.hexdigest()
+
+
+class ResolveRequirementsTaskBase(Task):
+  """Base class for tasks that resolve 3rd-party Python requirements.
+
+  Creates an (unzipped) PEX on disk containing all the resolved requirements.
+  This PEX can be merged with other PEXes to create a unified Python environment
+  for running the relevant python code.
+  """
+
+  @classmethod
+  def subsystem_dependencies(cls):
+    return super(ResolveRequirementsTaskBase, cls).subsystem_dependencies() + (PythonSetup, PythonRepos)
+
+  @classmethod
+  def prepare(cls, options, round_manager):
+    round_manager.require_data(PythonInterpreter)
+
+  def resolve_requirements(self, req_libs):
+    fs = PythonRequirementFingerprintStrategy(task=self)
+    with self.invalidated(req_libs, fingerprint_strategy=fs) as invalidation_check:
+      # If there are no relevant targets, we still go through the motions of resolving
+      # an empty set of requirements, to prevent downstream tasks from having to check
+      # for this special case.
+      if invalidation_check.all_vts:
+        target_set_id = VersionedTargetSet.from_versioned_targets(
+            invalidation_check.all_vts).cache_key.hash
+      else:
+        target_set_id = 'no_targets'
+
+      interpreter = self.context.products.get_data(PythonInterpreter)
+      path = os.path.join(self.workdir, str(interpreter.identity), target_set_id)
+      if not os.path.isdir(path):
+        path_tmp = path + '.tmp'
+        safe_rmtree(path_tmp)
+        self._build_pex(interpreter, path_tmp, req_libs)
+        safe_rmtree(path)
+        shutil.move(path_tmp, path)
+
+    return PEX(os.path.realpath(path), interpreter=interpreter)
+
+  def _build_pex(self, interpreter, path, req_libs):
+    builder = PEXBuilder(path=path, interpreter=interpreter, copy=True)
+
+    # Gather and de-dup all requirements.
+    reqs = OrderedSet()
+    for req_lib in req_libs:
+      for req in req_lib.requirements:
+        reqs.add(req)
+
+    # See which ones we need to build.
+    reqs_to_build = OrderedSet()
+    find_links = OrderedSet()
+    for req in reqs:
+      # TODO: should_build appears to be hardwired to always be True. Get rid of it?
+      if req.should_build(interpreter.python, Platform.current()):
+        reqs_to_build.add(req)
+        self.context.log.debug('  Dumping requirement: {}'.format(req))
```

Comment:
That was just copied from existing code, and I also think it is purposeful. I'll audit that this indenting makes sense.